From 2de5edcf8f4097ed68bff75ab6ba062f615be8fd Mon Sep 17 00:00:00 2001
Message-Id: <patch-2.2-2de5edcf8f4-20210411T144128Z-avarab@gmail.com>
In-Reply-To: <cover-0.2-00000000000-20210411T144128Z-avarab@gmail.com>
References: <20210411125431.28971-1-sir@cmpwn.com>
	<cover-0.2-00000000000-20210411T144128Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Wed, 7 Apr 2021 22:23:50 +0200
Subject: [PATCH 2/2] send-email: refactor sendemail.smtpencryption config
 parsing
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

With the removal of the support for sendemail.smtpssl in the preceding
commit the parsing of sendemail.smtpencryption is no longer special,
and can by moved to %config_settings.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 git-send-email.perl | 9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/git-send-email.perl b/git-send-email.perl
index 877c7dd1a21..da28c6e8b4b 100755
--- a/git-send-email.perl
+++ b/git-send-email.perl
@@ -268,6 +268,7 @@ sub do_edit {
 );
 
 my %config_settings = (
+    "smtpencryption" => \$smtp_encryption,
     "smtpserver" => \$smtp_server,
     "smtpserverport" => \$smtp_server_port,
     "smtpserveroption" => \@smtp_server_options,
@@ -368,14 +369,6 @@ sub read_config {
 			$$target = $v;
 		}
 	}
-
-	if (!defined $smtp_encryption) {
-		my $setting = "$prefix.smtpencryption";
-		my $enc = Git::config(@repo, $setting);
-		return unless defined $enc;
-		return if $configured->{$setting}++;
-		$smtp_encryption = $enc;
-	}
 }
 
 # sendemail.identity yields to --identity. We must parse this
-- 
2.31.1.623.g88b15a793d

